sdk/log/record.go

```diff
@@ -233,3 +233,20 @@ func (r *Record) Clone() Record {
 	res.back = slices.Clone(r.back)
 	return res
 }
+
+// TestRecord is a log record for testing.
+// You can use it for unit testing [Processor] and [Exporter] implementations.
+// Do not use TestRecord in production code.
+type TestRecord struct {
```

Comment:
This seems to be polluting the API with testing types.

It also is trying to use a `Test` prefix here to prevent semantic API use. I would rather explore options that use the semantics of Go actually make the restriction by design.

Reply:
I propose this as an intermediate solution as noted in the description.

Reply:
I would rather we work on an permanent and intentional solution instead of a stop gap.